Feature/follow up/map center

omotoshoayomikun opened the pull request:
This is a follow-up on what Darin implemented. I made a few adjustments to restructure the code a bit. Here is a breakdown of what I did:

1. Solved the build error.
2. Applied some styling to ensure the map fits the screen.
3. Restructured the code.

![Screenshot 2024-06-11 010520](https://github.com/map-of-pi/map-of-pi-frontend-react/assets/96895292/f106c2fb-5deb-488c-80fe-aae7d38af351)

DarinHajou committed:
Create the center-map Folder and page.tsx

DarinHajou committed:
Add content to page.tsx

DarinHajou committed:
Create CenterMap.tsx and add initial content

DarinHajou committed:
Add crosshair icon

DarinHajou committed:
Integrate leaflet for map functionality

DarinHajou committed:
Moved _document file inside page directory

DarinHajou committed:
Styled the CenterMap component

DarinHajou committed:
Enhance CenterMap component with popup message and dynamic crosshair

DarinHajou committed:
Add map of pi logo to public directory

DarinHajou committed:
Define icons for map center and saved location pins, and set up initial state and components

DarinHajou committed:
Add CenterMarker component to render the saved location pin

DarinHajou committed:
Add state management for marker position.

DarinHajou committed:
Update map event handling to enable and disable save button

DarinHajou committed:
Adjust logic for displaying and closing the welcome popup

DarinHajou committed:
Add styles for enabled and disabled button states

DarinHajou committed:
Handle visibility state for the "Save Center" button

DarinHajou committed:
Adjust crosshair icon size

DarinHajou committed:
Handle "save center" visibility states

DarinHajou committed:
Adjust "save center" button

DarinHajou committed:
Hide crosshair on save, show tooltip, and toggle save button visibility on map move

DarinHajou committed:
Modify typeText effect to correct interval typing and update welcome message

DarinHajou committed:
Modify "save center" button visibility

DarinHajou committed:
Update styles for CenterMarker popup message

DarinHajou committed:
Update styles for CenterMap.css

DarinHajou committed:
Move _document.tsx to pages directory

DarinHajou committed:
Move _document.tsx to pages folder

DarinHajou committed:
Change format from svg to jpeg

DarinHajou committed:
Change image format to jpeg

DarinHajou committed:
Remove comment

DarinHajou committed:
Change file name to _document.ts

DarinHajou committed:
Delete files

DarinHajou committed:
Fix build error

swoocn committed:
Resolve <img> warning from build.

swoocn committed:
Merge branch 'dev' into feature/set-map-center

swoocn committed:
Rename files to 'Map Center'.

swoocn committed:
Revert to initial text and keyframes.

omotoshoayomikun committed:
Build on what Darin implemented

swoocn committed:
Merge remote-tracking branch 'origin/dev' into feature/follow_up/map-center

swoocn commented:
Nice job @omotoshoayomikun!
Any chance you could resolve the warning resulting from the build?
 
![image](https://github.com/map-of-pi/map-of-pi-frontend-react/assets/72060658/63f33326-ecd1-41eb-b1df-1a50ae6572b8)

swoocn commented:
Could you also link the Map Center feature from the Seller Registration screen as well?

![image](https://github.com/map-of-pi/map-of-pi-frontend-react/assets/72060658/214d7ef3-6ce5-49a0-9b2a-a1987e621bc2)

swoocn committed:
Minor import adjustments.

omotoshoayomikun committed:
resolve warning resulting from the build

I resolve the warning resulting from the build.
I also link the Map Center feature from the Seller Registration screen as well

omotoshoayomikun committed:
Merge branch 'feature/follow_up/map-center' of https://github.com/map-of-pi/map-of-pi-frontend-react into feature/follow_up/map-center

swoocn commented on src/components/seller/add-edit-seller/SellerRegistration.tsx:
```diff
@@ -53,26 +60,22 @@ const SellerRegistrationForm = () => {
       </div>
-      <button className={`${BUTTON_CLASS} bg-green-600 text-white mb-4`}>Set Search Centre</button>
+      <div className="flex justify-end">
+      <Button label='Set Search Centre' onClick={() => router.push('/map-center')} />
```
Thoughts about using the `Link` component to handle navigation rather than router?
```
<Link href="/map-center">
  <Button label='Set Search Centre'/>
</Link>

```

swoocn reviewed (approved):
LGTM; great work with the feature! 😎👍
